docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

```diff
@@ -0,0 +1,508 @@
+
+This guide will walk you through the process of extending functionality for Explore queries in a data source plugin, enabling seamless integration and maximizing the potential of log analysis in your data source. By implementing these APIs, you can enhance the user experience and provide valuable insights through Explore's powerful log investigation capabilities.
+
+## Supported APIs for external plugin developers
+
+### Logs data frame format
+
+Here is the proposal for how should data frames for log results look like: https://grafana.github.io/dataplane/logs/.
+
+The data frame should include following fields:
+
+| Field name     | Field type                              | Info                                                                                                                                                                                                                                       |
+| -------------- | --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| **timestamp**  | `time`                                  | Field with timestamp, non nullable.                                                                                                                                                                                                        |
+| **body**       | `string`                                | Field with content of the log line content, non nullable.                                                                                                                                                                                  |
+| **severity**   | `string`                                | Represents the severity/level of the log line. If no severity field is found, consumers/client will decide the log level. More info about log level can be find [here](https://grafana.com/docs/grafana/latest/explore/logs-integration/). |
+| **id**         | `string`                                | Unique identifier of the log line.                                                                                                                                                                                                         |
+| **attributes** | `json raw message` (go) or `other` (ts) | This field represent additional attributes of the log line. Other systems may refer to this with other names, for example Loki calls these "labels". Value should be represented with Record<string,any> type in javascript.               |
+
+Logs data frame's `type` needs to be set to `type: DataFrameType.LogLines` in data frame's meta.
+
+Example of constructing a logs data frame in `go`:
+
+> Note: Should be used if you are creating data frames on backend in go code.
```

Comment:
```suggestion
Example of constructing a logs data frame in a `go` backend:
```
I would consider something like this. Or "go backend part of a plugin".

Reply:
I don't think I understand the comment. Could you please add more context. 

And just to clarify - developers can create data frames in backend and in frontend. Depending on where they are doing it, then go/typescript snippet is relevant.

Reply:
Oh yeah, makes sense! I'll update.